src/main/java/pl/zz/sharefood/controllers/FoodController.java

```diff
@@ -5,27 +5,39 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 import pl.zz.sharefood.domain.Food;
+import pl.zz.sharefood.domain.dto.FoodDto;
+import pl.zz.sharefood.payload.PagedResponse;
 import pl.zz.sharefood.repository.FoodRepository;
+import pl.zz.sharefood.service.FoodGetService;
 import pl.zz.sharefood.service.FoodService;
 import pl.zz.sharefood.service.FoodDeleteService;
+import pl.zz.sharefood.util.AppConstants;
 
 import java.util.List;
 
 @RestController
-@RequiredArgsConstructor
 @RequestMapping("/food")
 public class FoodController {
 
 
-    private final FoodRepository foodRepository;
 
     private final FoodService foodService;
 
-    final private FoodDeleteService foodDeleteService;
+    private final FoodGetService foodGetService;
+
+    private final  FoodDeleteService foodDeleteService;
+
+    FoodController(FoodDeleteService foodDeleteService, FoodGetService foodGetService, FoodService foodService){
+        this.foodDeleteService = foodDeleteService;
+        this.foodGetService = foodGetService;
+        this.foodService = foodService;
+    }
     
     @GetMapping
-    public List<Food> getAllFood() {
-        return foodRepository.findAll();
+    public PagedResponse<FoodDto> getAllFood(@RequestBody List<String> foodList,
+                                             @RequestParam(value = "page", defaultValue = AppConstants.DEFAULT_PAGE_NUMBER) int page,
```

Comment:
RequestParamy możesz przyjąć w formie obiektu

Reply:
Hmm, nie wiem, czy dobrze rozumiem, czy tak można to zrobić czyli np :
final class PageDetail{
   private int page;

I tutaj robić od razu walidację ?

Reply:
getAllFood(ListPageRequest request)

class ListPageRequest {
private int page;
itd.
}